### ISSUE TITLE
`dolfinx.fem.assemble_scalar` is not parallel safe

### ISSUE THREAD

**Issue by RemDelaporteMathurin** [festim-dev/FESTIM#896]
@jorgensd realised that this is not parallel safe and would return the wrong result in parallel.

https://github.com/festim-dev/FESTIM/blob/2677c3a31c62a91b940bbb1beb1dc18d53e3d2c4/festim/exports/derived_quantities/average_volume.py#L46-L48

Would be nice to have a demonstration of this

**Comment by hmonroedd** [comment]
should close this!
